packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx

```diff
@@ -27,29 +33,61 @@ const NotificationPaymentRecipient: React.FC<Props> = ({
   handleDownloadAttachamentPagoPA,
   handleReloadPayment,
 }) => {
-  const isSinglePayment = payments.length === 1;
+  const { pagoPaF24, f24Only } = payments;
+
+  const isSinglePayment = pagoPaF24.length === 1;
 
   const [selectedPayment, setSelectedPayment] = useState<PagoPAPaymentHistory | null>(null);
```

Comment:
Now if you have the payments already set, you can do useState<PagoPAPaymentHistory | null>(isSinglePayment ? ... : null);